src/main/java/org/apache/datasketches/quantiles/ItemsSketch.java

```diff
@@ -237,311 +258,369 @@ static <T> ItemsSketch<T> copy(final ItemsSketch<T> sketch) {
-    return getQuantile(fraction, false);
-  }
+  public Class<T> getSketchType() { return clazz; }
 
   /**
-   * Gets the upper bound of the value interval in which the true quantile of the given rank
-   * exists with a confidence of at least 99%.
-   * @param fraction the given normalized rank as a fraction
-   * @return the upper bound of the value interval in which the true quantile of the given rank
-   * exists with a confidence of at least 99%. Returns NaN if the sketch is empty.
+   * Returns the maximum item of the stream. This is provided for convenience, but may be different from the largest
+   * item retained by the sketch algorithm.
+   *
+   * <p>If the sketch is empty this returns null.</p>
+   *
+   * @return the maximum item of the stream
    */
-  public T getQuantileUpperBound(final double fraction) {
-    return getQuantile(min(1.0, fraction + Util.getNormalizedRankError(k_, false)));
-  }
+  public T getMaxItem() { return maxItem_; }
 
   /**
-   * Gets the lower bound of the value interval in which the true quantile of the given rank
-   * exists with a confidence of at least 99%.
-   * @param fraction the given normalized rank as a fraction
-   * @return the lower bound of the value interval in which the true quantile of the given rank
-   * exists with a confidence of at least 99%. Returns NaN if the sketch is empty.
+   * Returns the minimum quantile of the stream. This is provided for convenience, but is distinct from the smallest
+   * quantile retained by the sketch algorithm.
+   *
+   * <p>If the sketch is empty this returns null.</p>
+   *
+   * @return the minimum quantile of the stream
    */
-  public T getQuantileLowerBound(final double fraction) {
-    return getQuantile(max(0, fraction - Util.getNormalizedRankError(k_, false)));
+  public T getMinItem() { return minItem_; }
+
+  /**
+   * Same as {@link #getPMF(Object[], QuantileSearchCriteria) getPMF(splitPoints, INCLUSIVE)}
+   * @param splitPoints an array of <i>m</i> unique, monotonically increasing quantiles
+   * @return a PDF array of m+1 probability masses as doubles on the interval [0.0, 1.0).
+   */
+  public double[] getPMF(final T[] splitPoints) {
+    return getPMF(splitPoints, INCLUSIVE);
   }
 
   /**
-   * This is a more efficient multiple-query version of getQuantile().
+   * Returns an approximation to the Probability Mass Function (PMF) of the input stream
+   * as an array of double probability masses on the interval [0.0, 1.0],
+   * given a set of splitPoints.
+   * The sum of the probability masses in the returned array is always 1.0.
+   *
+   * <p>The resulting approximations have a probabilistic guarantee that can be obtained from the
+   * getNormalizedRankError(true) function.
+   *
+   * <p>If the sketch is empty this returns null.</p>
+   *
+   * @param splitPoints an array of <i>m</i> unique, monotonically increasing quantiles
+   * that divide the real number line into <i>m+1</i> consecutive non-overlapping intervals.
```

Comment:
this is generic sketch, so perhaps "input domain" or something of that sort instead of "real number line"

Reply:
This is now fixed.